src/Services/SearchService.php

```diff
@@ -472,4 +475,30 @@ private function buildItemURL($itemData, $withVariationId = true): string
             )->toRelativeUrl($includeLanguage);
         }
     }
+
+    /**
+     * @param Request $request
+     * @return mixed
+     */
+    private function requestWithRetry(Request $request)
+    {
+        for ($i = 0; $i <= self::MAX_RETRIES; $i++) {
+            $errorMsg = '';
+            $responseData = $this->client->call($request);
+            
+            if (is_array($responseData) && array_key_exists('error', $responseData)) {
+                $errorMsg = 'Plentymarkets returned error response';
+            } elseif (!is_string($responseData)) {
+                $errorMsg = 'Invalid response received from server';
+            }
```

Comment:
Wouldn't it make sense to move this to a separate method? E.g. `getErrorMessage`. This way we do not need to default to an empty string, which confused me when I initially read this, until I saw that we do an empty check at the end. This way we could just do a null check and `getErrorMessage` only returns an error if there is one, otherwise `null`.